src/liboslexec/shadingsys.cpp

```diff
@@ -2453,8 +2453,13 @@ ShadingSystemImpl::ShaderGroupEnd (ShaderGroup& group)
     ustring groupname = group.name();
     if (groupname.size() && groupname == m_archive_groupname) {
         std::string filename = m_archive_filename.string();
-        if (! filename.size())
+        if (! filename.size()) {
             filename = OIIO::Filesystem::filename (groupname.string()) + ".tar.gz";
+            // Colons will ruin our day when passed to tar as a filename.
+            // Why, yes, some people name their shader groups with colons.
+            if (OIIO::Strutil::contains(filename, ":"))
```

Comment:
Maybe catch `/`, `\` and `|` as well?

Reply:
The call to `Filesystem::filename` already strips out anything prior to the last directory separator. Hmmm, maybe watching out for `|` isn't a bad idea, though.